tests/test_tags.py

```diff
@@ -454,27 +441,23 @@ def test_linux_platforms_manylinux_unsupported(self, monkeypatch):
 
     def test_linux_platforms_manylinux1(self, is_x86, monkeypatch):
         monkeypatch.setattr(
-            tags, "_is_manylinux_compatible", lambda name, _: name == "manylinux1"
+            tags, "_is_manylinux_compatible", lambda name, *args: name == "manylinux1"
         )
-        if platform.system() != "Linux" or not is_x86:
-            monkeypatch.setattr(distutils.util, "get_platform", lambda: "linux_x86_64")
-            monkeypatch.setattr(platform, "machine", lambda: "x86_64")
-            monkeypatch.setattr(os, "confstr", lambda x: "glibc 2.20", raising=False)
+        monkeypatch.setattr(distutils.util, "get_platform", lambda: "linux_x86_64")
+        monkeypatch.setattr(platform, "machine", lambda: "x86_64")
+        monkeypatch.setattr(os, "confstr", lambda x: "glibc 2.20", raising=False)
         platforms = list(tags._linux_platforms(is_32bit=False))
         arch = platform.machine()
         assert platforms == ["manylinux1_" + arch, "linux_" + arch]
 
     def test_linux_platforms_manylinux2010(self, is_x86, monkeypatch):
-        monkeypatch.setattr(
-            tags, "_is_manylinux_compatible", lambda name, _: name == "manylinux2010"
-        )
-        if platform.system() != "Linux" or not is_x86:
-            monkeypatch.setattr(distutils.util, "get_platform", lambda: "linux_x86_64")
-            monkeypatch.setattr(platform, "machine", lambda: "x86_64")
-            monkeypatch.setattr(os, "confstr", lambda x: "glibc 2.20", raising=False)
+        monkeypatch.setattr(distutils.util, "get_platform", lambda: "linux_x86_64")
+        monkeypatch.setattr(platform, "machine", lambda: "x86_64")
+        monkeypatch.setattr(os, "confstr", lambda x: "glibc 2.12", raising=False)
         platforms = list(tags._linux_platforms(is_32bit=False))
         arch = platform.machine()
         expected = [
+            "manylinux_2_12_" + arch,
             "manylinux2010_" + arch,
             "manylinux_2_11_" + arch,
             "manylinux_2_10_" + arch,
```

Comment:
Now the expected values include the PEP 600 variant of the legacy tags, which is better.